front/src/modules/trainschedule/components/ManageTrainSchedule/helpers/updateTimetableItemHelper.ts

```diff
@@ -0,0 +1,171 @@
+  dispatch: AppDispatch,
+  timetableId: number,
+  pacedTrain: PacedTrain
+): Promise<TimetableItemWithTimetableId> {
+  const newPacedTrains = await dispatch(
+    osrdEditoastApi.endpoints.postTimetableByIdPacedTrains.initiate({
+      id: timetableId,
+      body: [pacedTrain],
+    })
+  ).unwrap();
+  return { ...newPacedTrains[0], id: formatEditoastTrainIdToPacedTrainId(newPacedTrains[0].id) };
+}
+
+async function updateTrainSchedule(
+  dispatch: AppDispatch,
+  id: TrainScheduleId,
+  trainSchedule: TrainScheduleBase
+) {
+  await dispatch(
+    osrdEditoastApi.endpoints.putTrainScheduleById.initiate({
+      id: formatTrainScheduleIdToEditoastTrainId(id),
+      trainScheduleForm: trainSchedule,
+    })
+  ).unwrap();
+}
+
+async function updatePacedTrain(dispatch: AppDispatch, id: PacedTrainId, pacedTrain: PacedTrain) {
+  await dispatch(
+    osrdEditoastApi.endpoints.putPacedTrainById.initiate({
+      id: formatPacedTrainIdToEditoastTrainId(id),
+      body: pacedTrain,
+    })
+  ).unwrap();
+}
+
+async function deleteTrainSchedule(dispatch: AppDispatch, id: TrainScheduleId): Promise<void> {
+  await dispatch(
+    osrdEditoastApi.endpoints.deleteTrainSchedule.initiate({
+      body: { ids: [formatTrainScheduleIdToEditoastTrainId(id)] },
+    })
+  ).unwrap();
+}
+
+async function deletePacedTrain(dispatch: AppDispatch, id: PacedTrainId): Promise<void> {
+  await dispatch(
+    osrdEditoastApi.endpoints.deletePacedTrain.initiate({
+      body: { ids: [formatPacedTrainIdToEditoastTrainId(id)] },
+    })
+  ).unwrap();
+}
+
+export async function storeTrainSchedule(
+  timetableItemIdToUpdate: TimetableItemId,
+  trainSchedule: TrainScheduleBase,
+  timetableId: number,
+  dispatch: AppDispatch,
+  upsertTimetableItems: (timetableItems: TimetableItemWithTimetableId[]) => void,
+  removeTimetableItems: (timetableItems: TimetableItemId[]) => void
+): Promise<TimetableItemWithTimetableId> {
+  if (isTrainSchedule(timetableItemIdToUpdate)) {
+    await updateTrainSchedule(dispatch, timetableItemIdToUpdate, trainSchedule);
+    const updatedTrainSchedule = {
+      ...trainSchedule,
+      id: timetableItemIdToUpdate,
+      timetable_id: timetableId,
+    };
+    upsertTimetableItems([updatedTrainSchedule]);
+    return updatedTrainSchedule;
+  }
+
+  // Turn a PacedTrain into a TrainSchedule
+  await deletePacedTrain(dispatch, timetableItemIdToUpdate);
+  const newTrainSchedule = await createTrainSchedule(dispatch, timetableId, trainSchedule);
+
+  removeTimetableItems([timetableItemIdToUpdate]);
+  upsertTimetableItems([newTrainSchedule]);
+  return newTrainSchedule;
+}
+
+export async function storePacedTrain(
+  timetableItemIdToUpdate: TimetableItemId,
+  pacedTrain: PacedTrain,
+  timetableId: number,
+  dispatch: AppDispatch,
+  upsertTimetableItems: (timetableItems: TimetableItemWithTimetableId[]) => void,
+  removeTimetableItems: (timetableItems: TimetableItemId[]) => void
+): Promise<TimetableItemWithTimetableId> {
```

Comment:
You can be more precise on the type here
```suggestion
): Promise<PacedTrainResponseWithPacedTrainId> {
```

You can do the same for storeTrainSchedule, createTrainSchedule and createPacedTrain

Reply:
Done in "front: precise typing in updateTimetableItemHelpers"